ocaml/xapi-storage-cli/main.ml

```diff
@@ -952,10 +945,15 @@ let cmds =
   ; mirror_start_cmd
   ; mirror_stop_cmd
   ]
+  |> List.map (fun (t, i) -> Cmd.v i t)
 
 let () =
-  match Term.eval_choice default_cmd cmds with
-  | `Error _ ->
-      exit 1
-  | _ ->
-      exit 0
+  let default =
+    Term.(ret (const (fun _ -> `Help (`Pager, None)) $ common_options_t))
+  in
+  let doc = "interact with an XCP storage management service" in
+  let info =
+    Cmd.info "sm-cli" ~version:"1.0.0" ~sdocs:_common_options ~doc ~man:help
```

Comment:
At some point we can fixup these versions to point to the true version, but for now lets keep it as is

Reply:
I am using this to get access to strings that we could use
```
! /bin/bash

cat <<EOF
let git_revision = "$(git describe --always --long)"
let build_time = "$(date)"
let build_timestamp = $(date +%s)
EOF

cat <<'EOF'
let json =
 `O [ "revision", `String git_revision
    ; "date", `String build_time
    ; "timestamp", `Float (float_of_int build_timestamp)
    ]
EOF
```
```
(rule
 (target build.ml)
 (deps (universe) build.sh)
 (action
  (with-stdout-to
   %{target}
   (bash "./build.sh"))))
```

Reply:
We have infrastructure to get version number in place (Build_info). It needs to be cleaned up so it doesn't conflict with dune's Build_info and the adopt it elsewhere.

Reply:
We should see whether we can use Dune's build-info support without requiring to define the `XAPI_VERSION` global, perhaps by having a separate module that parses the build info generated by dune. Dune has some support to avoid needless recompilations whenever the build info changes (it only embeds the true version number into the binary when it is installed, not when it is initially built/executed locally), but only if its own build info module is used.


Although that is a bit complicated because Dune's build info is designed NOT to be used at build time (it returns None there), but we do use version info during build time in gen_lifecycle.ml. With a few tweaks to the Makefile I think we can force these to use a (locally installed) version though, I'll try something and submit a PR to master if I succeed.